x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealmSettingsTests.java

```diff
@@ -0,0 +1,374 @@
+                final Settings settings = Settings.builder().put(settingKey, rejectedValue).build();
+                final Boolean actualValue = this.buildConfig(settings).getSetting(setting);
+                fail("No exception. Expected one for " + settingKey + "=" + rejectedValue + ". Got " + actualValue + ".");
+            });
+            assertThat(
+                exception.getMessage(),
+                equalTo("Failed to parse value [" + rejectedValue + "] as only [true] or [false] are allowed.")
+            );
+        }
+        for (final String ignoredValue : new String[] { null }) {
+            final Settings settings = Settings.builder().put(settingKey, ignoredValue).build();
+            final Boolean actualValue = this.buildConfig(settings).getSetting(setting);
+            assertThat(actualValue, equalTo(setting.getDefault(settings)));
+        }
+        for (final String acceptedValue : new String[] { "true", "false" }) {
+            final Settings settings = Settings.builder().put(settingKey, acceptedValue).build();
+            final Boolean actualValue = this.buildConfig(settings).getSetting(setting);
+            assertThat(actualValue, equalTo(Boolean.valueOf(acceptedValue)));
+        }
+    }
+
+    public void testClientAuthenticationType() {
+        Setting.AffixSetting<String> setting = JwtRealmSettings.CLIENT_AUTHENTICATION_TYPE;
+        final String settingKey = RealmSettings.getFullSettingKey(REALM_NAME, setting);
+        for (final String rejectedValue : new String[] { "", "unknown" }) {
+            final Exception exception = expectThrows(IllegalArgumentException.class, () -> {
+                final Settings settings = Settings.builder().put(settingKey, rejectedValue).build();
+                final String actualValue = this.buildConfig(settings).getSetting(setting);
+                fail("No exception. Expected one for " + settingKey + "=" + rejectedValue + ". Got " + actualValue + ".");
+            });
+            assertThat(
+                exception.getMessage(),
+                equalTo("Invalid value [" + rejectedValue + "] for [" + settingKey + "]." + " Allowed values are [sharedsecret, none]}].")
+            );
+        }
+        for (final String ignoredValue : new String[] { null }) {
+            final Settings settings = Settings.builder().put(settingKey, ignoredValue).build();
+            final String actualValue = this.buildConfig(settings).getSetting(setting);
+            assertThat(actualValue, equalTo(setting.getDefault(settings)));
+        }
+        for (final String acceptedValue : new String[] { "sharedsecret", "none" }) {
+            final Settings settings = Settings.builder().put(settingKey, acceptedValue).build();
+            final String actualValue = this.buildConfig(settings).getSetting(setting);
+            assertThat(actualValue, equalTo(acceptedValue));
+        }
+    }
+
+    public void testTimeSettingsWithDefault() {
```

Comment:
Do we need to test our TimeValue settings parser code in this class ?

Reply:
I included it for completeness over all JWT realm settings. We don't need to test TimeValue functionality, only test the completeness of our settings.